docs/explanations/advanced.md

```diff
@@ -1,158 +1,178 @@
+The API server copies that to the configured S3 storage for use during the staging of the application.
 
-When installing Epinio, the user can choose to use an external S3 compatible storage or let Epinio install either Minio or s3gw on the cluster ([See how to here](../howtos/customization/setup_external_s3.md)).
+When installing Epinio, you choose to use external S3 compatible storage, or let Epinio install one of Minio or s3gw on the cluster.
+You can see a [How-to here](../howtos/customization/setup_external_s3.md).
 
 ### s3gw
 
-[Upstream project link](https://github.com/aquarist-labs/s3gw)
+[The s3gw project link](https://github.com/aquarist-labs/s3gw)
 
-S3gw is a lightweight S3-compatible solution available in Epinio helm-chart as an alternative to Minio or external S3 provider.
+S3gw is a lightweight S3-compatible solution.
+You can specify it in the Epinio Helm chart as an alternative to Minio or an external S3 provider.
 
 ### Container Registry
 
-The result of Epinio's application staging is a container image. This image is used to create a Kubernetes deployment to run the application code.
-The staging Job requires that image to be written to some container registry (See also [Detailed push process](detailed-push-process.md)). 
+The result of Epinio's application staging is a container image.
+You use this image to create a Kubernetes deployment to run the application code.
+The staging job writes the image to a container registry.
+There is further information in the [detailed push process](detailed-push-process.md) documentation.
 
-By default the Epinio installation deploys a container registry inside the Kubernetes cluster to make the process easy and fast.
+By default, the Epinio installation deploys a container registry inside the Kubernetes cluster, making the process fast and convenient.
 
 Epinio comes with two consumers of this registry:
 
 1. Staging job - pushing the images
-2. Kubernetes - pulling the images when a Deployment is created for the Application
+2. Kubernetes - pulling the images when creating a deployment for the application
 
-Ideally all consumers will communicate with the registry over TLS to keep the communication encrypted.
-Epinio controls the staging job and ensures that whatever CA is used to sign the registry certificate is trusted by it. Achieving the same for Kubernetes however requires configuration that cannot happen from within the cluster, therefore Epinio has no way to ensure that. Theoretically there are 3 options:
+<!--TODO:Did I reword this correctly. I think it needs some better explanation. Who can help here?-->
+All consumers should communicate with the registry using TLS to encrypt communication.
+Epinio controls the staging job and it ensures that iit trusts the CA used to sign the registry certificate.
+Achieving the same for Kubernetes requires configuration that is impossible in the cluster,
```

Comment:
```suggestion
Achieving the same for Kubernetes requires configuration that is impossible from within the cluster,
```
```suggestion
Achieving the same for Kubernetes requires configuration that is impossible from inside the cluster,
```
I.e. an application like E, running in a pod of the cluster does not and cannot have the permissions to change things so highly relevant to cluster security. Such changes can only be made from the outside, by a cluster operator.

Reply:
Thanks.